### PR TITLE
Updated InlineInput

### PR DESCRIPTION
**Pull request by devconn99**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/form/Input/InlineInput/InlineInput.tsx [hunk 1]
```diff
@@ -40,22 +50,45 @@ export interface InlineInputProps
...
     },
     ref: Ref<HTMLInputElement>
   ) => {
     const theme: InputTheme = useComponentTheme('input', customTheme);
+    const [inputValue, setInputValue] = useState('');
+    const spanStyles =
+      'relative opacity-0 font-[inherit] text-[inherit] leading-none whitespace-pre select-none';
 
     return (
-      <AutosizeInput
```
Comment (amcdnl) [inline src/form/Input/InlineInput/InlineInput.tsx:53]: We can probably remove this depedency now.

File: src/form/Input/InlineInput/InlineInput.tsx [hunk 2]
```diff
@@ -40,22 +50,45 @@ export interface InlineInputProps
...
       theme: customTheme,
+      extraWidth,
+      onChange,
       ...rest
     },
     ref: Ref<HTMLInputElement>
   ) => {
     const theme: InputTheme = useComponentTheme('input', customTheme);
+    const [inputValue, setInputValue] = useState('');
+    const spanStyles =
```
Comment (amcdnl) [inline src/form/Input/InlineInput/InlineInput.tsx:66]: Need to test this on a variety of fonts and font sizes to ensure proper spacing.

Comment (ghsteff) [reply src/form/Input/InlineInput/InlineInput.tsx:60]: I tested on some diff fonts and sizes, lookin good 👍

Comment (ghsteff) [reply src/form/Input/InlineInput/InlineInput.tsx:60]: if `font` and `text` are inherited by default, do we need to include `font-[inherit] text-[inherit]`?

Comment (devconn99) [reply src/form/Input/InlineInput/InlineInput.tsx]: Removed font-[inherit] text-[inherit] leading-[inherit]

**Review by ghsteff — CHANGES_REQUESTED** [review]
ooop this affects `SelectInput` - the placeholder text and selections aren't visible anymore
image

File: src/form/Input/InlineInput/InlineInput.tsx [hunk 3]
```diff
@@ -40,22 +51,54 @@ export interface InlineInputProps
...
-        placeholderIsMinWidth={placeholderIsMinWidth}
-        {...rest}
-      />
+      <div className={twMerge(className, theme.inline.base)}>
+        <span className={spanStyles} style={{ paddingRight: extraWidth ?? 0 }}>
+          {inputValue}
+        </span>
+        {!placeholderIsMinWidth === true && !inputValue && (
+          <span className={spanStyles}>{placeholder}</span>
+        )}
```
Comment (ghsteff) [inline src/form/Input/InlineInput/InlineInput.tsx:92]: This `!placeholderIsMinWidth === true` seems backwards. Now a component like this has a width of 0 and the placeholder isn't visible

```
InlineInput
  placeholder="Type here..."
  placeholderIsMinWidth
/>
```

File: src/form/Input/InlineInput/InlineInput.tsx [hunk 4]
```diff
@@ -40,22 +51,54 @@ export interface InlineInputProps
...
+
+    const setRef = (node: HTMLInputElement) => {
+      ref?.current?.inputRef;
+
+      const placeholder = node?.placeholder;
+      const value = node?.value;
+
+      placeholder && setPlaceholderValue(placeholder);
+      value && setInputValue(value);
+    };
```
Comment (ghsteff) [inline src/form/Input/InlineInput/InlineInput.tsx:83]: is this a callback ref? Cool, I'm not familiar never used it. What's line 76 `ref?.current?.inputRef` doin?

If you pass a `ref` to `InlineInput` it doesn't get attached the actual input anymore right? So it's not behaving as you'd expect

File: src/form/Select/SelectInput/SelectInput.tsx [hunk 5]
```diff
@@ -4,6 +4,7 @@ import React, {
   Ref,
   RefObject,
   useCallback,
+  useEffect,
```
Comment (ghsteff) [inline src/form/Select/SelectInput/SelectInput.tsx:7]: this imports unused

**Comment by amcdnl** [comment]
@devconn99 - I think we are closing to having this merged, just a few minor tweaks.

**Comment by amcdnl** [comment]
@devconn99 - I believe we need to take a approach like this - https://dev.to/zechdc/autosize-input-field-plain-js-and-react-examples-38kb ... Are you interested in taking that on?

**Pull request closed without merging**